core/sequencer/server.go

```diff
@@ -151,35 +164,52 @@ func (s *Server) RunBatch(ctx context.Context, in *spb.RunBatchRequest) (*empty.
 		}
 	}
 
-	msgs, err := s.readMessages(ctx, in.DomainId, sources)
-	if err != nil {
-		return nil, err
+	//
+	// Rate limit the creation of new batches.
+	///
+
+	var totalCount int32
+	for _, count := range counts {
+		totalCount += count
 	}
-	if len(msgs) < int(in.MinBatch) {
-		return &empty.Empty{}, nil
+	// TODO(#1057): If time since last map revision > max timeout, run batch.
```

Comment:
Note: If we change batching strategy to consider per-log batch sizes, then the timeout should be checked on per-log basis as well.

Reply:
Perfect.